### ISSUE TITLE
Infinite loop with onInput

### ISSUE THREAD

**Issue by mahdix** [developergovindgupta/multi-range-slider-react-ts#9]
I'm using latest version (2.0.7). Looks like if I change a non-trivial state inside onInput handler, it causes infinite calls to the handler:

<img width="163" alt="image" src="https://github.com/developergovindgupta/multi-range-slider-react-ts/assets/9294049/2c311c1a-e502-46d6-92b6-184c7be87be9">

I simply modified the original codesandbox: [url](https://codesandbox.io/p/sandbox/demo-forked-67c673?file=%2Fsrc%2FApp.tsx%3A28%2C31&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clwzow3q300073b6kol2lzlcd%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clwzow3q300033b6kxmhe8z9j%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clwzow3q300043b6k8dsuci8b%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clwzow3q300063b6krumbwkhs%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clwzow3q300033b6kxmhe8z9j%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clwzow3q300023b6kt6m0rwi6%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A28%252C%2522startColumn%2522%253A31%252C%2522endLineNumber%2522%253A28%252C%2522endColumn%2522%253A31%257D%255D%257D%255D%252C%2522id%2522%253A%2522clwzow3q300033b6kxmhe8z9j%2522%252C%2522activeTabId%2522%253A%2522clwzow3q300023b6kt6m0rwi6%2522%257D%252C%2522clwzow3q300063b6krumbwkhs%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clwzow3q300053b6k55pb5fqs%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clwzow3q300063b6krumbwkhs%2522%252C%2522activeTabId%2522%253A%2522clwzow3q300053b6k55pb5fqs%2522%257D%252C%2522clwzow3q300043b6k8dsuci8b%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clwzow3q300043b6k8dsuci8b%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

Basically, my change is like:

```
  onInput={(e) => {
          setV({ ...v, min: e.minValue, max: e.maxValue });
          console.log("onInput called");
        }}
```
and it causes the loop;

If I use two separate state variables, everything works fine.
```
setMin(e.minValue);
setMax(e.maxValue);
```

**Comment by ullaskunder1** [comment]
I created a separate component and using that i have two slider component it is now re-rendering creating infinite loop

**Comment by pinokio221** [comment]
Same issue
